.github/workflows/node.js.yml

```diff
@@ -26,12 +26,14 @@ jobs:
       - name: Install pnpm
         uses: pnpm/action-setup@v2.0.1
         with:
-          version: 6.23.2
+          version: next
 
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
+          cache: 'pnpm'
+          cache-dependency-path: $${{ runner.os }}-node-${{ hasFiles('pnpm-lock.yaml') }}
```

Comment:
you may need `actions/cache@v2` if you want to use `hasFiles` function
https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time